codegen/src/main/twirl/templates/ScalaClient/Client.scala.txt

```diff
@@ -9,36 +9,23 @@ package @service.packageName
 
 import akka.NotUsed
 import akka.Done
-import akka.annotation.InternalApi
+import akka.grpc.GrpcClientSettings
 import akka.grpc.internal._
 import akka.grpc.scaladsl._
-import akka.grpc.GrpcClientSettings
-import akka.stream.{ Materializer, OverflowStrategy }
-import akka.stream.scaladsl.{ Flow, Sink, Source }
-import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Source
 import akka.stream.Materializer
 import scala.concurrent.{ ExecutionContext, Future, Promise }
 
 import io.grpc._
-import io.grpc.stub.{ ClientCalls, StreamObserver }
-import @{service.name}Client._
 
 final class @{service.name}Client(settings: GrpcClientSettings)(implicit mat: Materializer, ex: ExecutionContext) extends @{service.name} with AkkaGrpcClient {
   import @{service.name}Client._
 
-  private val internalChannel: InternalChannel = NettyClientUtils.createChannel(settings)
-  private val channel: Future[ManagedChannel] = internalChannel.managedChannel
-
   private val options = NettyClientUtils.callOptions(settings)
-
-  mat match {
-    case m: ActorMaterializer =>
-      m.system.whenTerminated.foreach(_ => close())(ex)
-    case _ =>
-  }
+  private val clientState = new ClientState(settings)
 
   @for(method <- service.methods) {
-    private val @{method.name}RequestBuilder = {
+    private def @{method.name}RequestBuilder(channel:Future[ManagedChannel]) = {
```

Comment:
```suggestion
    private def @{method.name}RequestBuilder(channel: Future[ManagedChannel]) = {
```

Reply:
There's a ton of (potential) formatting improvements in the code templates. I think we can sort them out at once in a separate PR.

Reply:
Sure.